### DISCUSSION TITLE
[Kavita] Allow setting a custom GitHub Personal Access Token for API calls

### DISCUSSION THREAD

**Discussion by gmahomarf** [Kareadita/Kavita#3271] [Ideas]
### Idea Description

I ran into an issue when trying out different themes. I hit the rate limit for the GitHub API, which is 60 requests per hour for unauthenticated users ([ref](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users)). It would be nice to be able to configure a GitHub PAT for authenticating with the API and increasing the limit.

### Idea Category

API

### Duration of Using Kavita

A little over a year

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
Hey gmahomarf, haven't seen you in a while. I never thought people would run out of rate. While adding a Github PAT is easy, I don't think it's the correct solution. Let me look into this a bit and see if I can leverage more caching on Kavita's side and Cloudflare to avoid this issue.

**Reply by gmahomarf** [reply]
Yeah, I haven't been using Kavita much lately for life reasons. Normally, I wouldn't have run out, but I was trying out different themes (dark is murder on my eyes) and reloading the page a few times. I had a bug that I couldn't figure out how to reproduce consistently where switching themes several times resulted in some styles being broken. That's where most of the API calls probably went.

Thanks for looking into this!

P.S. The beauty of open source is that I hardcoded my own PAT in the code, rebuilt and redeployed. Problem solved for now 😃

**Comment by ptninja** [comment]
Edit:
After reading another comment from [here](https://github.com/Kareadita/Kavita/issues/2852#issuecomment-2053696672), I tried to load the [URL](https://api.github.com/repos/Kareadita/Themes/contents/Native%20Themes) in browser and it opens w/o problem, then I can properly load the theme list. Not sure what is causing it.

Original comment:
I also ran into this, after upgrading from 0.8.2 to the latest `0.8.3.2`. I don't think I did any extensive call to github to begin with. This is what my theme settings page looks like: 

![image](https://github.com/user-attachments/assets/e13dd5d7-0192-4351-8bf7-6755526bec26)
